godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -457,6 +457,42 @@ macro_rules! impl_vector_fns {
     }
 }
 
+/// Implements functions that are present only on integer vectors.
+macro_rules! impl_integer_vector_fns {
+    (
+        // Names of the components, for example `x, y`.
+        $($comp:ident),*
+    ) => {
+        /// A new vector with each component snapped to the closest multiple of the corresponding
+        /// component in `step`.
+        pub fn snapped(self, step: Self) -> Self {
+            #[inline]
+            fn snap_one(mut value: i32, step: i32) -> i32 {
+                if step != 0 {
+                    let a = value + step / 2;
+
+                    // manual implement `a.div_floor(step)` since Rust's native method is still unstable, as of 1.79.0
+                    let mut d = a / step;
+                    let r = a % step;
```

Comment:
This can't [panic](https://doc.rust-lang.org/std/primitive.i32.html#panics-25) because if `a` is ever equal to `i32::MIN`(`-2147483648`) and `step` is `-1`, `value` needs to be `-2147483647.5` which is imposible.

Reply:
if `value` is `i32::MIN` and step is `-1` then `a` is `i32::MIN` since `-1 / 2 = 0` for `i32`.